src/utilities.jl

```diff
@@ -0,0 +1,109 @@
+"""
+    hasdups(guess::NTuple{N,Char}) where {N}
+
+Returns `true` if there are duplicate characters in `guess`.
+"""
+function hasdups(guess::NTuple{N,Char}) where {N}
+    @inbounds for i in 1:(N - 1)
+        gi = guess[i]
+        for j in (i + 1):N
+            gi == guess[j] && return true
+        end
+    end
+    return false
+end
+
+"""
+    scorecolumn!(col, guess::NTuple{N,Char}, targets::AbstractVector{NTuple{N,Char}})
+
+Return `col` updated with the scores of `guess` on each of the elements of `targets`
+
+If there are no duplicate characters in `guess` a simple algorithm is used, otherwise
+the more complex algorithm that accounts for duplicates is used.
+"""
+function scorecolumn!(
+    col::AbstractVector{<:Integer},
+    guess::NTuple{N,Char},
+    targets::AbstractVector{NTuple{N,Char}},
+) where {N}
+    if axes(col) ≠ axes(targets)
+        throw(DimensionMismatch("axes(col) = $(axes(col)) ≠ $(axes(targets)) = axes(targets)"))
+    end
+    if hasdups(guess)
+        svec = Vector{Int}(undef, N)         # scores for characters in guess
+        unused = trues(N)                    # has a character in targets[i] been used
+        @inbounds for i in axes(targets, 1)
+            sc = 0                           # integer value of score
+            targeti = targets[i]
+            for j in 1:N                 # first pass checking for target in same position
+                if guess[j] == targeti[j]
+                    unused[j] = false
+                    svec[j] = 2
+                else
+                    unused[j] = true
```

Comment:
you already initialize this to `true`

Reply:
It wasn't initialized within the `i` loop.  I have changed the logic now.  It is surprisingly difficult to get this correct.

Reply:
> It is surprisingly difficult to get this correct.

I feel much better about my eyes repeatedly crossing while reading this then. :smile: